packages/flutter/lib/src/widgets/nested_scroll_view.dart

```diff
@@ -498,7 +498,11 @@ class _NestedScrollCoordinator implements ScrollActivityDelegate, ScrollHoldCont
 
   bool get hasScrolledBody {
     for (_NestedScrollPosition position in _innerPositions) {
-      if (position.pixels > position.minScrollExtent)
+      if (
+        position.minScrollExtent != null &&
+        position.pixels != null &&
+        position.pixels > position.minScrollExtent
+      )
```

Comment:
nit: can you add `{}` around the `return true`?

Reply:
Also, I think we want a `TODO` on this with a link to the issue where we fix the multiple build/layout problem

Reply:
I think the crashing here itself is a bug with or without the multiple build problem. Do you think this is something we want to revert?

Reply:
I think we'd want to replace these with `assert(position.minScrollExtent != null)` and `assert(position.pixels != null)`.  If the code is designed correctly, this should never be legal.  If we keep this as is with no TODO, someone will look at it and think it's ok.

Reply:
I was under the impression that widget can build however many time without any dependency. If this is not the case, I will close this pr and fix the double build issue instead.

Reply:
I think this fix is worth pursuing temporarily while we figure out the double build issue - unless you think that can get done pretty quickly.  We've left this hanging around for a while with no good work around.

That said, the real fix is fixing the double build issue, and once that's done we'd want to replac these with asserts.

Reply:
ok, I rewritten the test to fit our intention.